src/srcset.js

```diff
@@ -231,4 +231,31 @@ export class Srcset {
   getLast() {
     return this.sources_[this.sources_.length - 1];
   }
+
+  /**
+   * Returns all sources in the srcset.
+   * @return {!Array<!SrcsetSourceDef>}
+   */
+  getSources() {
+    return this.sources_;
+  }
+
+  /**
+   * Reconstructs the string expression for this srcset.
+   * @return {string}
+   */
+  stringify() {
```

Comment:
I don't really like coercion to be honest. Usually I try to steer away. You feel strongly about it,  I'll switch.